tests/client/test_orion_client.py

```diff
@@ -1705,3 +1706,69 @@ async def test_delete_artifact_succeeds(self, orion_client, artifacts):
     async def test_delete_nonexistent_artifact_raises(self, orion_client):
         with pytest.raises(prefect.exceptions.ObjectNotFound):
             await orion_client.delete_artifact(uuid4())
+
+
+class TestVariables:
+    @pytest.fixture
+    async def variable(
+        self,
+        client,
+    ):
+        res = await client.post(
+            "/variables/",
+            json=schemas.actions.VariableCreate(
+                name="my_variable", value="my-value", tags=["123", "456"]
+            ).dict(json_compatible=True),
+        )
+        assert res.status_code == 201
+        return pydantic.parse_obj_as(schemas.core.Variable, res.json())
+
+    @pytest.fixture
+    async def variables(
+        self,
+        client,
+    ):
+        variables = [
+            schemas.actions.VariableCreate(
+                name="my_variable1", value="my-value1", tags=["1"]
+            ),
+            schemas.actions.VariableCreate(
+                name="my_variable2", value="my-value2", tags=["2"]
+            ),
+            schemas.actions.VariableCreate(
+                name="my_variable3", value="my-value3", tags=["3"]
+            ),
+        ]
+        results = []
+        for variable in variables:
+            res = await client.post(
+                "/variables/", json=variable.dict(json_compatible=True)
+            )
+            assert res.status_code == 201
+            results.append(res.json())
+        return pydantic.parse_obj_as(List[schemas.core.Variable], results)
+
+    async def test_read_variable_by_name(self, orion_client, variable):
+        res = await orion_client.read_variable_by_name(variable.name)
+        assert res.name == variable.name
+        assert res.value == variable.value
+        assert res.tags == variable.tags
+
+    async def test_read_variable_by_name_doesnt_exist(self, orion_client):
+        res = await orion_client.read_variable_by_name("doesnt_exist")
+        assert res is None
+
+    async def test_delete_variable_by_name(self, orion_client, variable):
+        await orion_client.delete_variable_by_name(variable.name)
+        res = await orion_client.read_variable_by_name(variable.name)
+        assert not res
```

Comment:
Perhaps also a test called `test_delete_variable_by_name_doesnt_exist`?

Reply:
woops missed that one, good call!